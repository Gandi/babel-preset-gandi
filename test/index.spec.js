const { transform } = require('@babel/core');
const preset = require('../');

let cases = [
  [
    'flow',
    `
      // @flow
      type A = string
      let a: A = 'hello'
    `,
  ],
  [
    'react',
    `
      let A = () => <div>hello</div>
    `,
  ],
  [
    'object rest spread',
    `
      let b = {...a}
    `,
  ],
  [
    'class properties',
    `
      class A { a = () => 'hello' }
    `,
  ],
  [
    'decorators legacy',
    `
      @decorated
      class A {}
    `,
  ],
  [
    'dynamic imports',
    `
      import('./a')
    `,
  ],
  [
    'Runtime ponyfills',
    `
      new Map()
      new Set()
      new Promise()
    `,
  ],
];

test.each(cases)('%s', (name, input) => {
  let { code } = transform(input, {
    presets: [preset],
  });

  expect(code).toMatchSnapshot();
});

test.each(cases)('commonjs: %s', (name, input) => {
  let { code } = transform(input, {
    presets: [[preset, { modules: 'commonjs' }]],
  });

  expect(code).toMatchSnapshot();
});
