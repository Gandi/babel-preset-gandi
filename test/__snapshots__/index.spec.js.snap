// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Runtime ponyfills 1`] = `
"import _Promise from \\"@babel/runtime-corejs2/core-js/promise\\";
import _Set from \\"@babel/runtime-corejs2/core-js/set\\";
import _Map from \\"@babel/runtime-corejs2/core-js/map\\";
new _Map();
new _Set();
new _Promise();"
`;

exports[`class properties 1`] = `
"import _classCallCheck from \\"@babel/runtime-corejs2/helpers/esm/classCallCheck\\";
import _defineProperty from \\"@babel/runtime-corejs2/helpers/esm/defineProperty\\";

var A = function A() {
  _classCallCheck(this, A);

  _defineProperty(this, \\"a\\", function () {
    return 'hello';
  });
};"
`;

exports[`commonjs: Runtime ponyfills 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime-corejs2/helpers/interopRequireDefault\\");

var _promise = _interopRequireDefault(require(\\"@babel/runtime-corejs2/core-js/promise\\"));

var _set = _interopRequireDefault(require(\\"@babel/runtime-corejs2/core-js/set\\"));

var _map = _interopRequireDefault(require(\\"@babel/runtime-corejs2/core-js/map\\"));

new _map.default();
new _set.default();
new _promise.default();"
`;

exports[`commonjs: class properties 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime-corejs2/helpers/interopRequireDefault\\");

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime-corejs2/helpers/classCallCheck\\"));

var _defineProperty2 = _interopRequireDefault(require(\\"@babel/runtime-corejs2/helpers/defineProperty\\"));

var A = function A() {
  (0, _classCallCheck2.default)(this, A);
  (0, _defineProperty2.default)(this, \\"a\\", function () {
    return 'hello';
  });
};"
`;

exports[`commonjs: decorators legacy 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime-corejs2/helpers/interopRequireDefault\\");

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime-corejs2/helpers/classCallCheck\\"));

var _class;

var A = decorated(_class = function A() {
  (0, _classCallCheck2.default)(this, A);
}) || _class;"
`;

exports[`commonjs: dynamic imports 1`] = `
"\\"use strict\\";

import('./a');"
`;

exports[`commonjs: flow 1`] = `
"\\"use strict\\";

var a = 'hello';"
`;

exports[`commonjs: object rest spread 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime-corejs2/helpers/interopRequireDefault\\");

var _objectSpread2 = _interopRequireDefault(require(\\"@babel/runtime-corejs2/helpers/objectSpread\\"));

var b = (0, _objectSpread2.default)({}, a);"
`;

exports[`commonjs: react 1`] = `
"\\"use strict\\";

var A = function A() {
  return React.createElement(\\"div\\", null, \\"hello\\");
};"
`;

exports[`decorators legacy 1`] = `
"import _classCallCheck from \\"@babel/runtime-corejs2/helpers/esm/classCallCheck\\";

var _class;

var A = decorated(_class = function A() {
  _classCallCheck(this, A);
}) || _class;"
`;

exports[`dynamic imports 1`] = `"import('./a');"`;

exports[`flow 1`] = `"var a = 'hello';"`;

exports[`object rest spread 1`] = `
"import _objectSpread from \\"@babel/runtime-corejs2/helpers/esm/objectSpread\\";

var b = _objectSpread({}, a);"
`;

exports[`react 1`] = `
"var A = function A() {
  return React.createElement(\\"div\\", null, \\"hello\\");
};"
`;
